recipes/cad1/task1/baseline/evaluate.py

```diff
@@ -0,0 +1,227 @@
+    audiogram_frequencies: np.ndarray,
+    fs_signal: int,
+) -> float:
+    """Compute HAAQI metric"""
+
+    haaqi_audiogram_freq = [250, 500, 1000, 2000, 4000, 6000]
+    audiogram_adjusted = np.array(
+        [
+            audiogram[i]
+            for i in range(len(audiogram_frequencies))
+            if audiogram_frequencies[i] in haaqi_audiogram_freq
+        ]
+    )
+    score, _, _, _ = haaqi_v1(
+        reference=ref_signal,
+        reference_freq=fs_signal,
+        processed=enh_signal,
+        processed_freq=fs_signal,
+        hearing_loss=audiogram_adjusted,
+        equalisation=1,
+    )
+    return score
+
+
+def set_song_seed(song: str) -> None:
+    """Set a seed that is unique for the given song"""
+    song_encoded = hashlib.md5(song.encode("utf-8")).hexdigest()
+    song_md5 = int(song_encoded, 16) % (10**8)
+    np.random.seed(song_md5)
+
+
+def make_song_listener_list(
+    songs: List[str], listeners: Dict[str, Any], small_test: bool = False
+) -> List[Tuple[str, str]]:
+    """Make the list of scene-listener pairing to process"""
+    song_listener_pairs = list(itertools.product(songs, listeners.keys()))
+
+    if small_test:
+        song_listener_pairs = song_listener_pairs[::15]
+
+    return song_listener_pairs
+
+
+@hydra.main(config_path="", config_name="config")
+def run_calculate_aq(cfg: DictConfig) -> None:
+    """Evaluate the enhanced signals using the HAAQI metric."""
+    # Load test songs
+    with open(cfg.path.valid_file, "r", encoding="utf-8") as fp:
+        songs = json.load(fp)
+    songs = pd.DataFrame.from_dict(songs)
+
+    # Load listener data
+    with open(cfg.path.listeners_file, "r", encoding="utf-8") as fp:
+        listener_audiograms = json.load(fp)
+
+    logger.info(f"Evaluating from {cfg.path.enhanced_folder} directory")
+    enhanced_folder = Path(cfg.path.enhanced_folder)
+
+    results_file = ResultsFile("scores.csv")
+    results_file.write_header()
+
+    song_listener_pair = make_song_listener_list(
+        songs["Track Name"].tolist(), listener_audiograms, cfg.evaluate.small_test
+    )
+
+    for song, listener in song_listener_pair:
+        logger.info(f"Evaluating {song} for {listener}")
+
+        if cfg.evaluate.set_random_seed:
+            set_song_seed(song)
+
+        scores = {}
+        for instrument in [
+            "drums",
+            "bass",
+            "other",
+            "vocals",
+        ]:
+            logger.info(f"...evaluating {instrument}")
+            split_dir = "train"
+            if songs[songs["Track Name"] == song]["Split"].tolist()[0] == "test":
+                split_dir = "test"
+
+            fs_ref_signal, ref_signal = wavfile.read(
+                Path(cfg.path.music_dir) / split_dir / song / f"{instrument}.wav"
+            )
+            ref_signal = ref_signal[30 * fs_ref_signal : 60 * fs_ref_signal, :]
+
+            ref_signal = (ref_signal / 32768.0).astype(np.float32)
+            l_ref_signal = ref_signal[:, 0]
```

Comment:
Abbreviations are common but its useful to be descriptive and use `left`/`right`/`sample_rate` in variable names rather than `l`/`r`/`fs`.